### ISSUE TITLE
nerdctl start container fails 2 of 3 times

### ISSUE THREAD

**Issue by dougbreaux** [rancher-sandbox/rancher-desktop#1600]
### Actual Behavior

When starting up an existing container, the first attempt gets one error, the second a different error, and the third succeeds. This has happened across multiple shells, for at least myself and a colleague. See [Slack thread](https://rancher-users.slack.com/archives/C0200L1N1MM/p1640101721002000).

### Steps to Reproduce

```console
nerdctl start <container-name>
```

### Result

```console
$ nerdctl start payment-tester

FATA[0000] OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: Running hook #0:: error running hook: exit status 1, stdout: , stderr: time="2022-01-15T02:38:19Z" level=fatal msg="failed to call cni.Setup: plugin type=\"bridge\" failed (add): failed to allocate for range 0: 10.4.0.9 has been allocated to default-5fad4b1feb5e5ed7fa2b7430d29bb5845943bb112cd0a1ce53543bde1f0eeba0, duplicate allocation is not allowed"

Failed to write to log, write /var/lib/nerdctl/dbb19c5e/containers/default/5fad4b1feb5e5ed7fa2b7430d29bb5845943bb112cd0a1ce53543bde1f0eeba0/oci-hook.createRuntime.log: file already closed: unknown

$ nerdctl start payment-tester

FATA[0000] OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: Running hook #0:: error running hook: exit status 1, stdout: , stderr: time="2022-01-15T02:38:25Z" level=fatal msg="failed to call cni.Setup: plugin type=\"bridge\" failed (add): failed to set bridge addr: could not set bridge's mac: invalid argument"

Failed to write to log, write /var/lib/nerdctl/dbb19c5e/containers/default/5fad4b1feb5e5ed7fa2b7430d29bb5845943bb112cd0a1ce53543bde1f0eeba0/oci-hook.createRuntime.log: file already closed: unknown

$ nerdctl start payment-tester

payment-tester
```

### Expected Behavior

Successful start

### Rancher Desktop Version

1.0.1

### Rancher Desktop K8s Version

1.21.7

### Which container runtime are you using?

containerd (nerdctl)

### What operating system are you using?

Windows

### Operating System / Build Version

Windows 10 Enterprise, Version 21H2 (OS Build 19044.1526)

### What CPU architecture are you using?

x64

**Comment by evertonlperes** [comment]
Thanks for filing the issue @dougbreaux 

Just for curiosity, 
Are you trying to execute the command right after the app loads? I mean, right after the `loading bar` disappears.
Just trying to understand if we have any racing conditions affecting the command execution.

Also, please don't share the Slack user channel link. The content are volatile and the message history will be gone in a few days. 
Feel free to copy and paste any relevant information from the Slack and post it into `Additional Information` field.

Thanks

**Comment by evertonlperes** [comment]
Did a couple of tests and I'd like to ask you if you can perform the same test case.

Here's what I did:
On PowerShell
```bash
# pulled nginx:latest and started the container
$ nerdctl stop nginx-[Id]
$ nerdctl ps # just to make sure the container is stopped
$ nerdctl start nginx-[Id] 
$ nginx-[id] # it started the container properly without error
```
Also, I did the same tests into the WSL (Leap 15.3) and I got the same results.

Do you mind to share your custom container or any other that has the same behaviour?

Thanks @dougbreaux

**Comment by dougbreaux** [comment]
@evertonlperes I can't share the custom container as it contains a proprietary web app. But it's based on OpenLiberty, so maybe some demo app with that can be used: 
```docker
FROM icr.io/appcafe/open-liberty:kernel-slim-java11-openj9-ubi
```
I'll also try an nginx as above.

**Comment by dougbreaux** [comment]
Well, either with plain nginx or plain open-liberty image above, stop/start worked fine.

So it either has to do with some additional Liberty setup in the Dockerfile, including the web app itself, or the additional command-line items I used when running the image. A volume mount of a key store file seeming the most likely possible culprit.

The original image run command was this:
```console
$ nerdctl run -d --name payment-tester-new --publish 9090:9080 --volume /Projects/saz/mvm-payment-test/trust.jks:/config/trust.jks --env TRUST_PASSWORD=${TRUST_PASSWORD} <private-repo>/payment-tester:3.7.3
```

**Comment by TMaYaD** [comment]
FWIW, this happens with stock `postgres` image too.

```sh
nerdctl run \
    --name postgres \
    --publish 5432:5432 \
    --env POSTGRES_PASSWORD=postgres \
    --detach \
    postgres
```

The error after restarting the machine:

```sh
± nerdctl start postgres
FATA[0000] failed to create shim: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: Running hook #0:: error running hook: exit status 1, stdout: , stderr: time="2022-04-25T01:44:24Z" level=fatal msg="failed to call cni.Setup: plugin type=\"bridge\" failed (add): failed to allocate for range 0: 10.4.0.9 has been allocated to default-1bf6325dbdb080a72e0ae869598f813fc2101c6ec63f9576158b72826adbbbdf, duplicate allocation is not allowed"
Failed to write to log, write /var/lib/nerdctl/dbb19c5e/containers/default/1bf6325dbdb080a72e0ae869598f813fc2101c6ec63f9576158b72826adbbbdf/oci-hook.createRuntime.log: file already closed: unknown
± nerdctl start postgres
FATA[0000] failed to create shim: OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: Running hook #0:: error running hook: exit status 1, stdout: , stderr: time="2022-04-25T01:44:31Z" level=fatal msg="failed to call cni.Setup: plugin type=\"bridge\" failed (add): failed to set bridge addr: could not set bridge's mac: invalid argument"
Failed to write to log, write /var/lib/nerdctl/dbb19c5e/containers/default/1bf6325dbdb080a72e0ae869598f813fc2101c6ec63f9576158b72826adbbbdf/oci-hook.createRuntime.log: file already closed: unknown
± ls /var/lib/nerdctl
ls: /var/lib/nerdctl: No such file or directory
± ls ~/Library/State/rancher-desktop/lima/rancher-desktop/*.log
zsh: no matches found: /Users/phoenix/Library/State/rancher-desktop/lima/rancher-desktop/*.log
± nerdctl start postgres
postgres
```

**Comment by xandris** [comment]
Ran into this with `mcr.microsoft.com/azure-sql-edge:latest`. Developer had a stopped container, rebooted their machine, tried to `nerdctl container start` the image, and encountered this. A second `nerdctl container start` worked, though.

```
% nerdctl container start azuresqledge
FATA[0000] 1 errors:
failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error running hook #0: error running hook: exit status 1, stdout: , stderr: time="2023-08-28T16:57:53Z" level=fatal msg="failed to call cni.Setup: plugin type=\"bridge\" failed (add): failed to allocate for range 0: 10.4.0.2 has been allocated to default-ef857d51b366775d37a6e41509c20c4c8b28be677ba7fafcaa8cfbdb1fb31570, duplicate allocation is not allowed"
Failed to write to log, write /var/lib/nerdctl/dbb19c5e/containers/default/ef857d51b366775d37a6e41509c20c4c8b28be677ba7fafcaa8cfbdb1fb31570/oci-hook.createRuntime.log: file already closed: unknown 
Error: exit status 1
```

**Comment by jandubois** [comment]
This issue is fixed in nerdctl 2.0, but we have to wait for a GA release.

**Comment by jandubois** [comment]
I've bumped `nerdctl` to `2.0.0-rc.3` now. This should be the last RC before GA. Hopefully GA will be released before Rancher Desktop 1.17; otherwise we'll ship with the RC3 build.
